dvc/state.py

```diff
@@ -455,7 +455,10 @@ def get(self, path_info):
             return None
 
         self._update_state_record_timestamp_for_inode(actual_inode)
-        return HashInfo("md5", value, size=int(actual_size))
+
+        mode = oct(os.stat(path).st_mode)
+
+        return HashInfo("md5", value, size=int(actual_size), mode=mode)
```

Comment:
mode is not really used by state, so we shouldn't include it here (we can probably do that in Output.save with self.tree.stat(self.path_info).st_mode. Also mode is not really related to HashInfo so it shoudn't be.a part of it, unless there are good reasons to include it. 🙂

Reply:
Yes, it makes sense not to include `mode` as part of hash. Should content of output file looks like:

```
outs:
- md5: acbd18db4cc2f85cedef654fccc4a4d8
  size: 3
  path: foo
- is_user_executable: false
```

So there will be two separate outputs. Will it affect anything that uses those files?

Could you provide a link to a place where `outs` are set in the code? I'm having trouble finding it quickly.

Reply:
@dudarev It should be more like:
```
outs:
- md5: acbd18db4cc2f85cedef654fccc4a4d8
  size: 3
  path: foo
  is_user_executable: false
```
Since that relates to the same output. Maybe you are confused about the way that entry is generated, see https://github.com/iterative/dvc/blob/f09c27e74c0a79c30c2c09a6b82201f4930ba63b/dvc/output/base.py#L299 . We can just add `ret[self.PARAM_ISEXEC] = self.isexec` there and parse it on the roundtrip similar to other fields.

> Could you provide a link to a place where outs are set in the code? I'm having trouble finding it quickly.

Sure! https://github.com/iterative/dvc/blob/f09c27e74c0a79c30c2c09a6b82201f4930ba63b/dvc/output/base.py#L289